app/controllers/herbarium_records_controller.rb

```diff
@@ -154,28 +131,37 @@ def edit
     @herbarium_record.herbarium_name = @herbarium_record.herbarium.try(&:name)
   end
 
-  alias_method :edit_herbarium_record, :edit
+  alias edit_herbarium_record edit
 
   def update
+    store_location
+    pass_query_params
+    @layout = calc_layout_params
+
+    @herbarium_record = find_or_goto_index(HerbariumRecord, params[:id])
+    return unless @herbarium_record
+
+    figure_out_where_to_go_back_to
+    return unless make_sure_can_edit!
+
     old_herbarium = @herbarium_record.herbarium
     @herbarium_record.attributes = whitelisted_herbarium_record_params
     normalize_parameters
-    if !validate_herbarium_name! ||
-       !can_add_record_to_herbarium?
-      return
-    elsif herbarium_label_free?
+
+    if !validate_herbarium_name! || !can_add_record_to_herbarium?
```

Comment:
This part of `:update` should be refactored, but I'm leaving that for now. There's a story that includes that refactoring.
https://www.pivotaltracker.com/story/show/173515917

It's complex and messy because the `if` includes bang methods, and the happy path is in the middle of the if/end code.